src/d3-ext/profile.js

```diff
@@ -0,0 +1,320 @@
+      x = d3.scale.linear().range([0, width]);
+
+      var height = this.getBoundingClientRect().height -
+          margin.top - margin.bottom;
+      y = d3.scale.linear().range([height, 0]);
+
+      var xAxis = d3.svg.axis().scale(x).orient('bottom');
+      var yAxis = d3.svg.axis()
+          .scale(y)
+          .orient('left');
+
+      var area = d3.svg.area()
+          .x(function(d) { return x(d['dist']); })
+          .y0(height)
+          .y1(function(d) { return y(d['values']['mnt']); });
+      var line = d3.svg.line()
+          .x(function(d) { return x(d['dist']); })
+          .y(function(d) { return y(d['values']['mnt']); });
+
+      // Select the svg element, if it exists.
+      svg = d3.select(this).selectAll('svg').data([data]);
+
+      // Otherwise, create the skeletal chart.
+      var gEnter = svg.enter().append('svg').append('g');
+      gEnter.style('font', '11px Arial');
+      gEnter.append('path').attr('class', 'area')
+          .style('fill', FILL_COLOR);
+      gEnter.append('path').attr('class', 'line')
+          .style('stroke', STROKE_COLOR)
+          .style('fill', 'none');
+
+      gEnter.insert('g', ':first-child')
+          .attr('class', 'grid-y');
+
+      gEnter.append('g')
+          .attr('class', 'x axis')
+          .attr('transform', 'translate(0,' + height + ')');
+
+      gEnter.append('text')
+          .attr('class', 'x label')
+          .attr('text-anchor', 'end')
+          .attr('x', width - 4)
+          .attr('y', height - 4);
+
+      gEnter.append('g')
+          .attr('class', 'y axis');
+
+      gEnter.append('text')
+          .attr('class', 'y label')
+          .attr('text-anchor', 'end')
+          .attr('y', 6)
+          .attr('dy', '.75em')
+          .attr('transform', 'rotate(-90)')
+          .style('fill', 'grey')
+          .text('elevation (m)');
+
+      gEnter.append('g')
+          .attr('class', 'metas')
+          .attr('transform', 'translate(' + (width + 3) + ', 0)');
+
+      var yHover = gEnter.append('g').attr('class', 'y grid-hover');
+      yHover.append('svg:line').attr('stroke-dasharray', '5,5');
+      yHover.append('text');
+
+      var xHover = gEnter.append('g').attr('class', 'x grid-hover');
+      xHover.append('svg:line').attr('stroke-dasharray', '5,5');
+      xHover.append('text');
+
+      gEnter.append('rect')
+          .attr('class', 'overlay')
+          .attr('width', width)
+          .attr('height', height)
+          .style('fill', 'none')
+          .style('pointer-events', 'all');
+
+      // Update the outer dimensions.
+      svg
+          .attr('width', width + margin.left + margin.right)
+          .attr('height', height + margin.top + margin.bottom);
+
+      // Update the inner dimensions.
+      var g = svg.select('g')
+          .attr('transform', 'translate(' + margin.left + ',' +
+              margin.top + ')');
+
+      var xDomain = d3.extent(data, function(d) { return d['dist']; });
```

Comment:
If an extractor as defined is https://github.com/camptocamp/ngeo/pull/173/files#r25346757 is used, the extractor method can be passed directly here: `d3.extent(data, extractor.getDistance);`